### PR TITLE
vpa: Pod Monitors for prometheus metrics

### PR DESCRIPTION
**Pull request by digiserg**
**Why This PR?**
I want to get the VPA metrics in proemtheus

Fixes #

**Changes**
Changes proposed in this pull request:

* adds optional podMonitor configuration

**Checklist:**

* [x] I have included the name of the chart in the title of this PR in square brackets i.e. `[stable/goldilocks]`.
* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or `helm-docs --sort-values-order=file` has been run for the charts that support it.

### CODE REVIEW

File: stable/vpa/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: vpa
 description: A Helm chart for Kubernetes Vertical Pod Autoscaler
 type: application
-version: 2.0.0
+version: 2.0.1
```
Comment (sudermanjr) [inline stable/vpa/Chart.yaml]:
```suggestion
version: 2.1.0
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]
Hello, thanks for the PR! 

I recommended a different version change due to the fact that we are adding functionality (according to semver that's a minor version change). Also, could you please add documentation strings to the new values that are being added?

File: stable/vpa/templates/recommender-podmonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{{- if and .Values.recommender.enabled .Values.recommender.podMonitor }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: {{ include "vpa.fullname" . }}-recommender
```
Comment (sudermanjr) [inline stable/vpa/templates/recommender-podmonitor.yaml:6]: Do we need to add the ability to add labels to the pod monitor? If I remember correctly, the prometheus operator uses those to select which pod monitors to pull in.

Comment (digiserg) [reply stable/vpa/templates/recommender-podmonitor.yaml:6]: It's a valid point. I'll add them.

Comment (sudermanjr) [reply stable/vpa/templates/recommender-podmonitor.yaml:6]: Thanks!

File: stable/vpa/values.yaml [hunk 3]
```diff
@@ -67,6 +67,7 @@ recommender:
   nodeSelector: {}
   tolerations: []
   affinity: {}
+  podMonitor: false
```
Comment (sudermanjr) [inline stable/vpa/values.yaml]:
```suggestion
  # -- Enables a prometheus operator podMonitor for the recommender
  podMonitor: false
```

File: stable/vpa/values.yaml [hunk 4]
```diff
@@ -105,6 +106,7 @@ updater:
...
```
Comment (sudermanjr) [inline stable/vpa/values.yaml]:
```suggestion
  # -- Enables a prometheus operator podMonitor for the updater
  podMonitor: false
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]
Docs strings. You'll need to run `helm-docs --sort-values-order=file` to have them update the readme.

**Comment by sudermanjr** [comment]
Seems like we need https://github.com/helm/chart-testing/pull/531 to be merged so that we can start disabling that. We're definitely failing at the reuse-values step right now

File: stable/vpa/templates/recommender-podmonitor.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{{- if and .Values.recommender.enabled .Values.recommender.podMonitor.enabled }}
```
Comment (sudermanjr) [inline stable/vpa/templates/recommender-podmonitor.yaml:2]: I believe wrapping these files in this if statement will fix the e2e tests for now until chart-testing allows us to skip that check.

```suggestion
{{- if .Values.recommender.podMonitor }}
{{- if and .Values.recommender.enabled .Values.recommender.podMonitor.enabled }}
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]

**Review by sudermanjr — APPROVED** [review]

**Comment by sudermanjr** [comment]
LGTM. Thanks!

**Pull request merged**